<a href="https://colab.research.google.com/github/pdpranavkumar/lecture1/blob/master/Play_Store_App_Review_Analysis_Pranav_Dhabale_Capstone_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> The Play Store apps data has enormous potential to drive app-making businesses to success. Actionable insights can be drawn for developers to work on and capture the Android market. </b>

## <b> Each app (row) has values for catergory, rating, size, and more. Another dataset contains customer reviews of the android apps.</b>

## <b> Explore and analyze the data to discover key factors responsible for app engagement and success. </b>

#Will Start with implementing the procedures by importing libraries.


In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline


#First we have to mount the data  store. 

In [3]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


#Reading the csv file


In [4]:
path="/content/drive/MyDrive/Almabetter/Play Store Data.csv"
play_store_df = pd.read_csv(path)

In [5]:
play_store_df.head(10)

App  ...   Android Ver
0     Photo Editor & Candy Camera & Grid & ScrapBook  ...  4.0.3 and up
1                                Coloring book moana  ...  4.0.3 and up
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ...  4.0.3 and up
3                              Sketch - Draw & Paint  ...    4.2 and up
4              Pixel Draw - Number Art Coloring Book  ...    4.4 and up
5                         Paper flowers instructions  ...    2.3 and up
6            Smoke Effect Photo Maker - Smoke Editor  ...  4.0.3 and up
7                                   Infinite Painter  ...    4.2 and up
8                               Garden Coloring Book  ...    3.0 and up
9                      Kids Paint Free - Drawing Fun  ...  4.0.3 and up

[10 rows x 13 columns]

#Check the information about the dataframe.

In [6]:
play_store_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


#Have a look on column

In [7]:
play_store_df.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='object')

#Check weather the dataframe contains the null value or not


In [8]:
play_store_df.isnull().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

#we can define function to view soome insights of dataset

In [13]:
def printinfo():
  temp = pd.DataFrame(index = play_store_df.columns)
  temp['data_type'] = play_store_df.dtypes
  temp['null_count'] = play_store_df.isnull().sum()
  temp['unique_count'] = play_store_df.nunique()
  return temp

In [14]:
printinfo()

data_type  null_count  unique_count
App               object           0          9660
Category          object           0            34
Rating           float64        1474            40
Reviews           object           0          6002
Size              object           0           462
Installs          object           0            22
Type              object           1             3
Price             object           0            93
Content Rating    object           1             6
Genres            object           0           120
Last Updated      object           0          1378
Current Ver       object           8          2832
Android Ver       object           3            33

#We can check rows of the column Rating having null values


In [18]:
play_store_df[play_store_df.Rating.isnull()]

App  ... Android Ver
23                  Mcqueen Coloring pages  ...  4.1 and up
113              Wrinkles and rejuvenation  ...  3.0 and up
123                 Manicure - nail design  ...  4.1 and up
126           Skin Care and Natural Beauty  ...  4.1 and up
129    Secrets of beauty, youth and health  ...  2.3 and up
...                                    ...  ...         ...
10824                            Cardio-FR  ...  4.4 and up
10825                   Naruto & Boruto FR  ...  4.0 and up
10831             payermonstationnement.fr  ...  4.0 and up
10835                             FR Forms  ...  4.0 and up
10838               Parkinson Exercices FR  ...  2.2 and up

[1474 rows x 13 columns]

#We can check rows of the column type having null values

In [19]:
play_store_df[play_store_df.Type.isnull()]

App  ...         Android Ver
9148  Command & Conquer: Rivals  ...  Varies with device

[1 rows x 13 columns]

#we can see their is one missing value in type column

#lets fill the missing value,crosschecked the missing value found to be free,so can fill it free

In [20]:
play_store_df['Type'].fillna("Free", inplace = True)

In [21]:
play_store_df.isnull().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 0
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

#we can also check rows of column content rating having null value

In [23]:
play_store_df[play_store_df['Content Rating'].isnull()]

App Category  ...  Current Ver Android Ver
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9  ...   4.0 and up         NaN

[1 rows x 13 columns]

In [24]:
play_store_df.loc[10468:10477, :]

App  ...   Android Ver
10468                         Tassa.fi Finland  ...    4.0 and up
10469              TownWiFi | Wi-Fi Everywhere  ...    4.2 and up
10470                               Jazz Wi-Fi  ...    2.3 and up
10471                         Xposed Wi-Fi-Pwd  ...  4.0.3 and up
10472  Life Made WI-Fi Touchscreen Photo Frame  ...           NaN
10473                  osmino Wi-Fi: free WiFi  ...    4.4 and up
10474                             Sat-Fi Voice  ...    2.2 and up
10475                         Wi-Fi Visualizer  ...    2.3 and up
10476                    Lennox iComfort Wi-Fi  ...  2.3.3 and up
10477              Sci-Fi Sounds and Ringtones  ...    4.0 and up

[10 rows x 13 columns]

#from the above command we see the row 10472 having missing data in category column,better is we can drop this row from our dataframe

In [25]:
play_store_df.dropna(subset = ['Content Rating'], inplace=True)

#we can drop unwanted columns 

In [26]:
play_store_df.drop(['Current Ver','Last Updated','Android Ver'], axis=1, inplace=True)

In [27]:
play_store_df.head()

App  ...                     Genres
0     Photo Editor & Candy Camera & Grid & ScrapBook  ...               Art & Design
1                                Coloring book moana  ...  Art & Design;Pretend Play
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ...               Art & Design
3                              Sketch - Draw & Paint  ...               Art & Design
4              Pixel Draw - Number Art Coloring Book  ...    Art & Design;Creativity

[5 rows x 10 columns]

#let us replace the missing value of Rating column with the mode value of entire column

In [28]:
modeValueRating = play_store_df['Rating'].mode()

In [33]:
modeValueRating[0]

4.4

In [35]:
play_store_df['Rating'].fillna(value=modeValueRating[0], inplace= True)

In [36]:
printinfo()

data_type  null_count  unique_count
App               object           0          9659
Category          object           0            33
Rating           float64           0            39
Reviews           object           0          6001
Size              object           0           461
Installs          object           0            21
Type              object           0             2
Price             object           0            92
Content Rating    object           0             6
Genres            object           0           119

#we are done with the data cleansing part and in state to start work for data preparation

Very first we need to convert data types with their respective datatypes

#Column Reviews

In [37]:
play_store_df['Reviews'] = play_store_df.Reviews.astype(int)

In [38]:
printinfo()

data_type  null_count  unique_count
App               object           0          9659
Category          object           0            33
Rating           float64           0            39
Reviews            int64           0          6001
Size              object           0           461
Installs          object           0            21
Type              object           0             2
Price             object           0            92
Content Rating    object           0             6
Genres            object           0           119

#Column Size

Removing the symbols

In [39]:
play_store_df['Size'] = play_store_df.Size.apply(lambda x: x.strip('+'))

In [41]:
play_store_df['Size'] = play_store_df.Size.apply(lambda x: x.replace(',',','))

Replacing the M by multiplying the value with 1000000

In [42]:
play_store_df['Size'] = play_store_df.Size.apply(lambda x: x.replace('M','e+6'))

In [45]:
play_store_df.head(5)

App  ...                     Genres
0     Photo Editor & Candy Camera & Grid & ScrapBook  ...               Art & Design
1                                Coloring book moana  ...  Art & Design;Pretend Play
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ...               Art & Design
3                              Sketch - Draw & Paint  ...               Art & Design
4              Pixel Draw - Number Art Coloring Book  ...    Art & Design;Creativity

[5 rows x 10 columns]

replacing k by multiplying the value with 1000

In [47]:
play_store_df['Size'] = play_store_df.Size.apply(lambda x: x.replace('k','e+3'))

In [48]:
play_store_df.head()

App  ...                     Genres
0     Photo Editor & Candy Camera & Grid & ScrapBook  ...               Art & Design
1                                Coloring book moana  ...  Art & Design;Pretend Play
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ...               Art & Design
3                              Sketch - Draw & Paint  ...               Art & Design
4              Pixel Draw - Number Art Coloring Book  ...    Art & Design;Creativity

[5 rows x 10 columns]

#replacing values with varies with device value to Nan

In [51]:
play_store_df['Size'] = play_store_df.Size.replace('Varies with device',np.NaN)

last converting to numeric

In [52]:
play_store_df['Size'] = pd.to_numeric(play_store_df['Size'])

In [53]:
printinfo()

data_type  null_count  unique_count
App               object           0          9659
Category          object           0            33
Rating           float64           0            39
Reviews            int64           0          6001
Size             float64        1695           459
Installs          object           0            21
Type              object           0             2
Price             object           0            92
Content Rating    object           0             6
Genres            object           0           119
Sixe              object        1695           460

In [54]:
play_store_df.head()

App  ...    Sixe
0     Photo Editor & Candy Camera & Grid & ScrapBook  ...   19e+6
1                                Coloring book moana  ...   14e+6
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ...  8.7e+6
3                              Sketch - Draw & Paint  ...   25e+6
4              Pixel Draw - Number Art Coloring Book  ...  2.8e+6

[5 rows x 11 columns]

In [55]:
play_store_df.dropna(subset= ['Size'], inplace=True)

In [56]:
printinfo()

data_type  null_count  unique_count
App               object           0          8434
Category          object           0            33
Rating           float64           0            39
Reviews            int64           0          4680
Size             float64           0           459
Installs          object           0            20
Type              object           0             2
Price             object           0            87
Content Rating    object           0             6
Genres            object           0           116
Sixe              object           0           460

#Column: installs